crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -527,6 +527,27 @@ impl Analyzer<'_, '_> {
 
                             let _context = debug_ctx!(format!("Property type: {}", dump_type_as_string(&prop_ty)));
 
+                            if let Type::IndexedAccessType(prop_ty) = prop_ty.normalize() {
+                                match prop_ty.index_type.normalize() {
+                                    // if you want remove this, first you should overflow narrow any
+                                    Type::Param(..)
+                                    | Type::Keyword(KeywordType {
+                                        kind:
+                                            TsKeywordTypeKind::TsNumberKeyword
+                                            | TsKeywordTypeKind::TsStringKeyword
+                                            | TsKeywordTypeKind::TsSymbolKeyword,
+                                        ..
+                                    })
+                                    | Type::Lit(LitType {
+                                        lit: RTsLit::Str(..) | RTsLit::Number(..),
+                                        ..
+                                    }) => {}
+                                    _ => {
+                                        panic!("{:?}", prop_ty);
```

Comment:
I'll not merge this PR while this is fixed

Reply:
May I ask what you are concerned about?

Reply:
It's simply a regression